repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,174 @@
+from spack_repo.builtin.build_systems.cmake import CMakePackage
+from spack_repo.builtin.build_systems.cuda import CudaPackage
+from spack_repo.builtin.build_systems.rocm import ROCmPackage
+
+from spack.package import *
+
+
+class Erf(CMakePackage, CudaPackage, ROCmPackage):
+    """ERF solves the compressible Navier-Stokes on a Arakawa C-grid
+    for large-scale weather modeling.
+    """
+
+    homepage = "https://erf.readthedocs.io/en/latest/index.html"
+    url = "https://github.com/erf-model/ERF/archive/refs/tags/25.08.tar.gz"
+    git = "https://github.com/erf-model/ERF.git"
+    test_requires_compiler = True
+
+    maintainers("larenspear")
+
+    license("BSD-3-Clause", checked_by="larenspear")
+
+    version("25.10", tag="25.10", submodules=True)
+    version("25.08", tag="25.08", submodules=True)
+    version("25.07", tag="25.08", submodules=True)
+    version("25.06", tag="25.08", submodules=True)
+    version("25.05", tag="25.08", submodules=True)
+    version("25.04", tag="25.08", submodules=True)
+    version("25.03", tag="25.08", submodules=True)
+    version("25.01", tag="25.08", submodules=True)
+    version("24.11", tag="25.08", submodules=True)
+    version("24.10", tag="25.08", submodules=True)
+    version("24.09", tag="25.08", submodules=True)
+    version("24.08", tag="25.08", submodules=True)
+    version("24.06", tag="25.08", submodules=True)
+    version("24.05", tag="25.08", submodules=True)
+    version("24.04", tag="25.08", submodules=True)
+    version("24.03", tag="25.08", submodules=True)
+    version("24.02", tag="25.08", submodules=True)
+    version("24.01", tag="25.08", submodules=True)
+    version("23.12", tag="25.08", submodules=True)
+    version("23.11", tag="25.08", submodules=True)
+    version("23.10", tag="25.08", submodules=True)
+
+    variant("mpi", default=False, description="Enable MPI support")
+    variant("openmp", default=False, description="Enable OpenMP support")
+    variant("hip", default=False, description="Enable HIP/rocm support")
```

Comment:
The `RocmPackage` has an implicit `rocm` variant that you should use instead of adding this one.